src/Esprima/Ast/Literal.cs

```diff
@@ -55,11 +48,19 @@ public Literal(string pattern, string flags, object? value, string raw) : this(T
     public string Raw { [MethodImpl(MethodImplOptions.AggressiveInlining)] get; }
     public RegexValue? Regex { [MethodImpl(MethodImplOptions.AggressiveInlining)] get; }
 
-    public string? StringValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.StringLiteral ? Value as string : null; }
-    public bool BooleanValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.BooleanLiteral && ReferenceEquals(Value, _boxedTrue); }
-    public double NumericValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get; }
+    public string? StringValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.StringLiteral ? (string) Value! : null; }
+    public bool BooleanValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.BooleanLiteral && ReferenceEquals(Value, s_boxedTrue); }
```

Comment:
note own my own doings, the TokenType check for BooleanValue is quite redundant

Reply:
As a matter of fact, `public bool? BooleanValue { [MethodImpl(MethodImplOptions.AggressiveInlining)] get => TokenType == TokenType.BooleanLiteral ? ReferenceEquals(Value, _boxedTrue) : null; }` might be a bit cleaner. It would be a BC but this way it'd be clear whether the literal has a boolean value or not. I'm leaning towards this change. What do you think?

Reply:
Well from Jint development perspective, we already check the token type to know which property to read value from, so there we know type = BooleanLiteral before accessing the getter so everything else but plain `true/false` is just noise. But I understand the idea behind missing boolean value as this is not a boolean. Currently there's no fast way to get just the reference equals as the boxed value is hidden. But this probably is in the category of micro-optimizations. I'm fine with either way. Jint would just do `BooleanValue.Value` blindly as type is known.

Reply:
Ok, then I'd say let's do this: https://github.com/sebastienros/esprima-dotnet/pull/310/commits/5184ee1a9f07246408b592d4b92b1726779d2e9d

> Jint would just do BooleanValue.Value blindly as type is known

When you know the type beforehand, you don't need the `XXXValue` properties at all, you can just do the cast directly. E.g. `(bool) Value` in the case of a boolean.

Otherwise, you can read `BooleanValue` and from the result it will be clear whether it actually has a boolean value (`true`/`false`) or not (`null`).